### PR TITLE
Easycore-0.3.0

### PR DESCRIPTION
**Pull request by arm61**
This PR gets EasyCore-0.3.0 being used in the library in particular the new MultiFitter. Also 3.10 has been added to the test CI and a lot of linting.

### CODE REVIEW

**Comment by arm61** [comment]
@wardsimon or @rozyczko can you have a look at this before I merge?

File: pyproject.toml [hunk 1]
```diff
@@ -28,17 +28,17 @@ classifiers = [
 packages = [ { include = "EasyReflectometry" } ]
 
 [tool.poetry.dependencies]
-python = "^3.8,<3.10"
-easyScienceCore = { git = 'https://github.com/easyScience/easyCore.git', rev = 'mutlifitter' }
+python = "^3.8,<3.11"
+easyScienceCore = { git = 'https://github.com/easyScience/easyCore.git', rev = 'pre-release' }
```
Comment (rozyczko) [inline pyproject.toml:32]: this is now merged into master, consider replacing

File: EasyReflectometry/special/calculations.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def neutron_scattering_length(formula: str) -> complex:
 def molecular_weight(formula: str) -> float:
     """
     Determine the molecular weight for a chemical formula.
-    
+
```
Comment (rozyczko) [inline EasyReflectometry/special/calculations.py:43]: Any reason to not use `periodictable` Formula object, with built-in parsers and molecular weight method?

File: EasyReflectometry/sample/layer.py [hunk 3]
```diff
@@ -372,38 +374,39 @@ def chemical_structure(self) -> str:
...
         layerapm_dict = super()._dict_repr
         layerapm_dict['chemical_structure'] = self._chemical_structure
         layerapm_dict[
-            'area_per_molecule'] = f'{self.area_per_molecule.raw_value:.1f} {self.area_per_molecule.unit}'
+            'area_per_molecule'] = f'{self.area_per_molecule.raw_value:.1f} ' \
+                                   f'{self.area_per_molecule.unit}'
         return layerapm_dict
 
-    def as_dict(self, skip: list=[]) -> dict:
+    def as_dict(self, skip: list = []) -> dict:
```
Comment (rozyczko) [inline EasyReflectometry/sample/layer.py:403]: Are you sure with defaulting `list` to `[ ]`? Lists are mutable default arguments and can be tricky.
https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument

**Review by rozyczko — APPROVED** [review]
Minor comments and questions.

**Pull request merged**